src/RevProc.cc

```diff
@@ -995,7 +995,11 @@ RevInst RevProc::DecodeRInst(uint32_t Inst, unsigned Entry){
   }
 
   // imm
-  DInst.imm     = 0x0;
+  if( (InstTable[Entry].imm == FImm) && (InstTable[Entry].rs2Class == RegUNKNOWN)){
+    DInst.imm  = DECODE_IMM12(Inst) & 0b011111; 
+  }else{
+    DInst.imm     = 0x0;
+  }
 
```

Comment:
Handle the special R-type that has the shift amount (this seems like the best place to do this?)